third-party/json/test/json11-test.cpp

```diff
@@ -0,0 +1,124 @@
+        }
+        SECTION("int64_t - min")
+        {
+            int64_t value = INT64_MIN;
+            auto json = json11::Json::object{
+                    {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("int32_t - max")
+        {
+            int32_t value = INT32_MAX;
+            auto json = json11::Json::object{
+                    {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("int32_t - min")
+        {
+            int32_t value = INT32_MIN;
+            auto json = json11::Json::object{
+                    {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("int16_t - min")
+        {
+            int16_t value = INT16_MAX;
+            auto json = json11::Json::object{
+                    {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("int16_t - max")
+        {
+            int16_t value = INT16_MIN;
+            auto json = json11::Json::object{
+                    {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("int8_t - min")
+        {
+            int8_t value = INT8_MAX;
+            auto json = json11::Json::object{
+                    {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("int8_t - max")
+        {
+            int8_t value = INT8_MIN;
+            auto json = json11::Json::object{
+                    {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+    }
+
+    SECTION("unsigned"){
+        SECTION("uintmax_t")
+        {
+            uintmax_t value = UINTMAX_MAX;
+            auto json = json11::Json::object{
+                     {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("uint64_t")
+        {
+            uint64_t value = UINT64_MAX;
+            auto json = json11::Json::object{
+                    {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("uint32_t")
+        {
+            uint32_t value = UINT32_MAX;
+            auto json = json11::Json::object{
+                    {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("uint16_t")
+        {
+            uint16_t value = UINT16_MAX;
+            auto json = json11::Json::object{
+                    {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("uint8_t")
+        {
+            uint8_t value = UINT8_MAX;
+            auto json = json11::Json::object{
+                   {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+        SECTION("size_t")
+        {
+            size_t value = SIZE_MAX;
+            auto json = json11::Json::object{
+                   {"test", value}};
+            REQUIRE(json["test"] == value);
+        }
+    }
```

Comment:
It might be worth to add test case like that:
create json object -> dump it to a string -> recreate json object from dumped string -> check if object fields are valid